lib/manager/pip_requirements/update.js

```diff
@@ -7,10 +7,7 @@ function updateDependency(fileContent, upgrade) {
     logger.debug(`pip_requirements.updateDependency(): ${upgrade.newValue}`);
     const lines = fileContent.split('\n');
     const oldValue = lines[upgrade.lineNumber];
-    const newValue = oldValue.replace(
-      /^.+?(\s.*)?$/,
-      `${upgrade.depName}${upgrade.newValue}$1`
-    );
+    const newValue = oldValue.replace(upgrade.currentValue, upgrade.newValue);
```

Comment:
This will fail in some circumstances. e.g. let's say currentValue is `==1.0.0` and there is an open PR to upgrade to `==1.0.1`. Next, `1.1.0` is released. The above substitution will fail because it will try to replace `1.0.0` with `1.1.0` but actually the branch's `requirements.txt` will contain `1.0.1`.

Reply:
We instead need to replace "whatever the current value is in this line";

Reply:
Isn't extract and update ran on the same file content?